### PR TITLE
[WIP] school detail and school list feature

### PR DESCRIPTION
**Pull request by SuppenGeist**
see #42 
Implement the school details view

### CODE REVIEW

**Comment by codingfabi** [comment]
I think it is almost completed... I hope we can finish it tomorrow, so stay tuned.

**Comment by Lorsbyjm** [comment]
Please also look at sentry. There was an error raised in your files

**Comment by TheSlimvReal** [comment]
> 
> 
> What's the status on this? Is this anywhere near completion?
> The social workers have some new schools to add now in December/January. If this still takes a while, maybe I will implement a simplified version for immediate use.

The SchoolDetail screen is finished, but now I also wanted to bring the new ChildSchoolRelations into play. This takes me a little bit more time but I think it would be nice to finally have this implemented.

File: src/app/entity/EntityRelation.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import {Entity} from './entity';
+
+export abstract class EntityRelation extends Entity {
```
Comment (sleidig) [inline src/app/entity/EntityRelation.ts]: In which way is `EntityRelation` different from `Entity`? Does it really require a separate class for this?

File: src/app/schools/schoolsShared/schools.services.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import {Injectable} from '@angular/core';
+import {School} from './school';
+import {EntityMapperService} from '../../entity/entity-mapper.service';
+import {Observable} from '../../../../node_modules/rxjs/Rx';
+
+@Injectable()
+export class SchoolsServices {
```
Comment (sleidig) [inline src/app/schools/schoolsShared/schools.services.ts]: Naming convention is to use "service" in singular (i.e. file name "schools.service.ts", class name "SchoolsService")

**Comment by sleidig** [comment]
I do understand why this is taking long - quite a complex challenge with the relationship modelling between `Child` and `School`.

Have you looked into solutions that keep the relationship loading out of the entity classes (i.e. move/remove `Child.getSchools()` and `School.getStudents()`)? Maybe having that code in the ChildrenService and SchoolService - or in the EntityMapper?
To move those Entity classes around, save them to the database - and first and foremost to keep them simple - I like the idea that Entity subtype classes only contain business logic. The idea was initially to let an EntityMapper class take care of all the loading, saving and also the resolution of relationships.

**Comment by sleidig** [comment]
I am taking out the completed features of the school details (everything up to a278a7d8a4c52fe2a5cc4007d1cbd86ca00af7ce ) into a separate PR: #204. So that we can make this available to users within the next days already.

**Pull request closed without merging**